pkg/tuner/tuner.go

```diff
@@ -0,0 +1,94 @@
+package tuner
+
+import (
+	"fmt"
+	"net/url"
+	"os/exec"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/vshn/waf-tool/cfg"
+	"github.com/vshn/waf-tool/pkg/elasticsearch"
+	"github.com/vshn/waf-tool/pkg/forwarder"
+	"github.com/vshn/waf-tool/pkg/model"
+	"github.com/vshn/waf-tool/pkg/rules"
+)
+
+// Tune creates exclusion rules for a given uniqe ID
+func Tune(uniqueID string, config cfg.Configuration) (returnError error) {
+	out, err := exec.Command("oc", "whoami", "--show-token").Output()
+	if err != nil {
+		return fmt.Errorf("could not get token: %w", err)
+	}
+	token := strings.TrimSpace(string(out))
+	es, err := elasticsearch.New(config.ElasticSearch, token)
+	if err != nil {
+		return fmt.Errorf("could not create ES client: %w", err)
+	}
+
+	url, err := url.Parse(config.ElasticSearch.URL)
+	if err != nil {
+		return err
+	}
+
+	fwd := forwarder.New(url.Port())
+
+	defer func() {
+		if err := fwd.Stop(); err != nil {
+			if returnError == nil {
+				returnError = err
+			} else {
+				// Already encountered an error so we log this one here
+				log.WithError(err).Error("Could not stop port forwarding")
+			}
+		}
+	}()
+	if err := fwd.Start(); err != nil {
+		return err
+	}
+
+	log.WithFields(log.Fields{
+		"url": config.ElasticSearch.URL,
+	}).Debug("Connecting to Elasticsearch...")
+
+	result, err := es.SearchUniqueID(uniqueID)
+	if err != nil {
+		return err
+	}
+
+	if len(result.Hits.Hits) == 0 {
+		log.WithField("unique-id", uniqueID).Warnf("Could not find any request")
+		return nil
+	}
+
+	baseID := 10100
```

Comment:
this seems to be hardcoded alone in the wind. Make it configurable via flags/env or at least as variable on top?

Reply:
Agree. I think in the future this needs to be configurable or it should even be set automatically based on other rules. For now I'll declare it as a constant.